### PR TITLE
Fix `validate_matching_label_keys`

### PR DESCRIPTION
**Pull request by czaloom**
`validate_matching_label_keys` was accessing data that was out of scope. This PR implements a hard boundary on the datasets and model involved in an evaluation.

### CODE REVIEW

File: api/valor_api/backend/core/label.py [hunk 1]
```diff
@@ -26,71 +26,95 @@ def validate_matching_label_keys(
...
             ),
         )
-        .select_from(gts)
-        .group_by(gts.c.datum_id)
+        .select_from(models.Datum)
+        .join(
+            models.Dataset,
+            and_(
+                models.Dataset.id == models.Datum.dataset_id,
+                models.Dataset.name.in_(dataset_names),
```
Comment (ntlind) [inline api/valor_api/backend/core/label.py:54]: could the filters be an important addition here? is there an edge case where selecting all labels from a dataset/model wouldn't pass this validation, but filtering the labels using some user's filters would pass this validation?

File: api/valor_api/backend/core/label.py [hunk 2]
```diff
@@ -26,71 +26,95 @@ def validate_matching_label_keys(
...
         Optional mapping of individual labels to a grouper label. Useful when you need to evaluate performance using labels that differ across datasets and models.
 
-
     Raises
     -------
     ValueError
         If the distinct ground truth label keys don't match the distinct prediction label keys for any datum.
     """
 
-    gts = generate_select(
```
Comment (ntlind) [inline api/valor_api/backend/core/label.py:43]: we use `generate_select` statements like this one in a few other places; could those be leaking data, too?

Comment (czaloom) [reply api/valor_api/backend/core/label.py:43]: Im reverting to using filters per your previous comment. These should not leak as it is now adding datasets, model, label_mapping and task_type to the select generation.

These were missing previously.

File: api/valor_api/backend/core/label.py (hunk 1 above)
Comment (czaloom) [reply api/valor_api/backend/core/label.py:54]: Ill revert to using `from valor_api.backend.metrics.metric_utils import prepare_filter_for_evaluation` as it properly formats the filter for evaluation.

File: client/valor/client.py [hunk 3]
```diff
@@ -273,36 +273,36 @@ def _requests_wrapper(
...
         """
         Helper for handling POST requests.
         """
         return self._requests_wrapper(
-            method_name="post", endpoint=endpoint, *args, **kwargs
+            method_name="post", endpoint=endpoint, **kwargs
         )
 
-    def _requests_get_rel_host(self, endpoint: str, *args, **kwargs):
+    def _requests_get_rel_host(self, endpoint: str, *_, **kwargs):
```
Comment (ntlind) [inline client/valor/client.py:284]: can you explain these "*args" -> "*_" changes a bit more? is this related to `validate_matching_label_keys` somehow?

File: api/valor_api/backend/core/label.py (hunk 1 above)
Comment (ntlind) [reply api/valor_api/backend/core/label.py]: thank you!

File: api/valor_api/backend/core/label.py (hunk 2 above)
Comment (ntlind) [reply api/valor_api/backend/core/label.py]: great, thanks!

File: client/valor/client.py (hunk 3 above)
Comment (ntlind) [reply client/valor/client.py:284]: it seems like your client and integration changes are unrelated to this PR, and may be leaked here from another PR. just want to confirm that's true. best practice would be to remove them, but I'd consider it a nit

Comment (czaloom) [reply client/valor/client.py]: Removed! They got dragged in by an unrelated pr.

**Review by ntlind — APPROVED** [review]

**Pull request merged**